Grammar Updates to the General Folder (#1026)
AdguardTeam/AdguardKnowledgeBase: 01.general/11.https-filtering/docs.en.md

The diagram below describes the difference between plain HTTP protocol and secure HTTPS protocol.

![What is HTTPS](https://cdn.adguard.com/public/Adguard/Blog/https/what_is_https.png)

### What is a security certificate?

Simply put, HTTPS means data encryption. But there’s still a problem: how can you be sure you’ve established an encrypted connection with exactly the website you intended to? That’s where security certificates come into play. The certificate is proof that the website is actually what it tells you it is. If a website doesn’t have such a certificate, or if the certificate contains incorrect information, the browser won’t let you establish a secure connection. It is important that the certificate a website uses is issued by a certification authority (CA) trusted by your browser. Such a CA guarantees that the SSL certificate is, indeed, issued to the website’s owner.

### Why does AdGuard need to be able to filter HTTPS?

> The thing is, more and more websites, blogs, social media, etc., switch to HTTPS every day. And following blogs and websites, more ad networks are switching to HTTPS too, because it becomes necessary to display ads on HTTPS-based websites. Here are a few examples of popular websites where you can’t remove ads without filtering the HTTPS: youtube.com, facebook.com, and twitter.com.

Author: ```suggestion
The thing is, as more and more websites, blogs, social media, and others switch to HTTPS, more ad networks follow them to HTTPS as well, because it becomes necessary to display ads on HTTPS-based websites. Here are a few popular websites where you can’t remove ads without HTTPS filtering: youtube.com, facebook.com, and twitter.com.
```

### How does HTTPS filtering work?

If it were easy, HTTPS wouldn’t be that secure. Upon a browser’s attempt to connect to a server, AdGuard establishes two secure connections: one with the browser (or another app), and the other with the server. It is crucial that the browser trusts AdGuard and the connections created by it. For this purpose, AdGuard generates a special (and unique) root certificate and installs it into the system and, when it is required, into some browsers too (e.g., Firefox). Thus, AdGuard can see what is happening inside of the secure connection and do its job — block ads and trackers.

For better understanding we depicted this process:

![How does HTTPS filtering work](https://cdn.adguard.com/public/Adguard/Blog/https/what_is_https_filtering.png)

### Does my traffic remain encrypted and secure?

Of course! Your connection with a remote server remains encrypted and secure. AdGuard, just like your browser, checks the server’s certificate before deciding whether to filter it or not.

> Nevertheless, HTTPS filtering has its drawbacks. The most important of these is the fact that they hide from the browser the real certificate that the website uses. Instead, the browser sees the certificate issued by AdGuard.

Author: ```suggestion
However, HTTPS filtering has its drawbacks. The most important one is that it hides from the browser the actual certificate of the website. Instead, the browser sees the certificate issued by AdGuard.
```
